feat: Updating import in MockERC20 to ERC20 instead of Openzeppelin

RafaDSan opened the pull request:
closes #92 

I updated the code in the file `MockERC20.sol` to import the `ERC20.sol` from this [repo](https://github.com/blockful-io/blockful-contracts/blob/main/contracts/token/ERC20/ERC20.sol).

I also added three new files to test the updated MockERC20, using as a base this [repo](https://github.com/blockful-io/swaplace-contracts/tree/main/test).

0xneves commented:
Hey @RafaDSan, thank you for the contribution. I think I left it a little bit confusing in the issue when referring to testing for bugs and issues, I was referring to issues in the Swaplace protocol and not in the ERC20 contract itself. 

The tests you copied already exist and there is no need to have them in the swaplace repository. 
And besides this first misunderstanding, there is another feature I would like to ask: we should not mix the erc20 contract with the Swaplace contracts, avoiding any sort of confusion for the devs by following a more reasonable approach.

Proposal:
- Delete the ERC20 tests.
- Add the ERC20 files inside the mock folder as in `contracts/mock/ERC20`.
- Adapt the file paths
- Test the system using `yarn test`

0xneves commented on contracts/interfaces/IERC20.sol:
move to contracts/mock/ERC20/interfaces

0xneves commented on contracts/interfaces/IERC20Errors.sol:
move to contracts/mock/ERC20/interfaces

0xneves commented on contracts/interfaces/IERC20Permit.sol:
move to contracts/mock/ERC20/interfaces

0xneves commented on contracts/mock/MockERC20.sol:
Adapt the file paths to the proposed path

0xneves commented on test/TestERC20Approval.test.ts:
Delete file

0xneves commented on test/TestERC20Supply.test.ts:
Delete file

0xneves commented on test/TestERC20Transfer.test.ts:
Delete file

RafaDSan commented:
> > Hey @RafaDSan, thank you for the contribution. I think I left it a little bit confusing in the issue when referring to testing for bugs and issues, I was referring to issues in the Swaplace protocol and not in the ERC20 contract itself.
> > The tests you copied already exist and there is no need to have them in the swaplace repository. And besides this first misunderstanding, there is another feature I would like to ask: we should not mix the erc20 contract with the Swaplace contracts, avoiding any sort of confusion for the devs by following a more reasonable approach.
> > Proposal:
> > 
> > * Delete the ERC20 tests.
> > * Add the ERC20 files inside the mock folder as in `contracts/mock/ERC20`.
> > * Adapt the file paths
> > * Test the system using `yarn test`
> 
> Requesting changes on each file. Overall description above.

Thank you very much for clarifying things to me! 

I did as you proposed and:

- [x] Deleted the ERC20 tests
- [x] Added the ERC20 files inside a `contracts/mock/ERC20/interfaces` file
- [x] Adpated the file paths in `MockERC20.sol`
- [x] Tested the contract using `yarn test` and it all worked fine.
